src/scimPatch.ts

```diff
@@ -247,9 +247,16 @@ function navigate(inputSchema: any, paths: string[]): any {
  */
 function addOrReplaceAttribute(property: any, patch: ScimPatchAddReplaceOperation): any {
     if (Array.isArray(property)) {
-        const isReplace = patch.op.toLowerCase() === 'replace';
-        if(isReplace && Array.isArray(patch.value))
+        if (Array.isArray(patch.value)) {
+            // if we're adding an array, we need to remove duplicated values from existing array
+            if (patch.op.toLowerCase() === "add") {
+                const existingItemValues = property.map(item => item.value)
+                const valuesToAdd = patch.value.filter(item => !existingItemValues.includes(item.value))
```

Comment:
Since `value` is not a mandatory parameter in the SCIM RFC (see **`addresses`** field in the RFC https://tools.ietf.org/html/rfc7643#section-8.2 for example).

I would suggest checking the full object and not only the `value` field.
What do you think about that @alicial?
```suggestion
                const valuesToAdd = patch.value.filter(item => !property.includes(item))
```